src/dicomweb_client/protocol.py

```diff
@@ -0,0 +1,717 @@
+from typing import (
+    Any,
+    Dict,
+    Iterator,
+    List,
+    Optional,
+    Protocol,
+    runtime_checkable,
+    Sequence,
+    Tuple,
+    Union,
+)
+
+from pydicom.dataset import Dataset
+
+
+@runtime_checkable
+class DICOMwebProtocol(Protocol):
```

Comment:
nit: thoughts on naming it `DICOMClient` or something similar? Protocols tend to be named after capabilities rather than their type.

Reply:
Addressed with ec880f926ab62b935567b93825ab23087bcc2a16